components/state-less-components.md

```diff
@@ -93,28 +96,75 @@ class MainPage : Component
 }
 ```
 
-We could create a component like this:
+{% endcode %}
+
+Run it with F5.  You should see a VStack of Label, Entry, Label, Entry, then a HStack of two buttons
+But the observant developer notices that we are repeating VStack(Label, Entry) twice
 
-<pre class="language-csharp"><code class="lang-csharp"><strong>partial class EntryWithLabel : Component
-</strong>{
+**DRY it out by adding our first custom Component**
+
+ 1. Create `Components` folder
+
+ 2. Add `Components/GlobalUsings.cs` with content:
+
+    ```
+    global using MauiReactor;
+    ```
+
+    By adding `global using`, we can save typing for all Component classes in Components folder.
+    We don't want to add it to root `GlobalUsings.cs` as we want it focused on the `Components` folder, and not be used in every class in the project.
+
+3. Add Components/EntryWithLabel.cs
+
+{% code lineNumbers="true" %}
+
+```
+namespace StatelessComponents.Components;
+
+partial class EntryWithLabel : Component
+{
     [Prop]
     string _labelText;
 
-    public override VisualNode Render()
-      => VStack(
+    public override VisualNode Render() =>
+        VStack(
             Label(_labelText),
             Entry()
         );
+
 }
-</code></pre>
+```
 
-and reuse it on the main page as shown in the following code:
+{% endcode %}
+
+- A MauiReactor code generator will refactor this code for you behind the scenes.
+  - There are **several rules** to follow when coding`[Prop] string _labelText;`
+    - `[Prop]` attribute is required to make the **field** a parameter.  
+      *This may be one of the rare times you must use a **field** rather than a property in your C# experience.*
+    - `[Prop]` will not compile for properties with `{ get; set; }`
+    - `[Prop] string _labelText = "will be ignored";` will compile, run and be ignored by the code generator, so don't do it.  
+      That would add **dead code** which needs to be reviewed and read by current and future developers.
+    - **Never** name your `_variable` as a Component name, such as `[Prop] string _label;`
+      Doing so will break the project with an infinite loop, never seeing your page render.
+    - The component `class` must be `partial` and inherit from `Component`.
+    - The `[Prop]` name must start with underscore and be `_camelCase`.
+    - Within the Component, use `_labelText` throughout as the example does.
+    - Each property that will be passed in has the `[Prop]` attribute.
+    - Each `[Prop]` should skip the access part such as `public`/`protected`/`internal`/`private`, which will be ignored by the MauiReactor code generator.
```

Comment:
It's not ignored but preserved, for example, it's perfectly valid to create a prop protected if you want to access it from a derived component class